firefoxThemes/resources.js

```diff
@@ -63,6 +74,27 @@ async function startStorage() {
   // Register all styles from option page
   updateOptions({optionName: "textSelection", optionValue: !!!storage.textSelection});
   updateOptions({optionName: "scrollbar", optionValue: !!!storage.scrollbar});
+  // Register to follow system color theme
+  if (storage.systemTheme === systemStates.SYSTEM || storage.systemTheme === systemStates.DRUTHERS) {
+    updateSystemListener({addObserver:true});
+    browser.browserSettings.overrideContentColorScheme.set({value: "system"});
+  } else {
+    browser.browserSettings.overrideContentColorScheme.set({value: "browser"});
+  }
+
+}
+/*Update the system observer to listen to system color theme changes*/
+function updateSystemListener(msg) {
+  if (msg.addObserver) {
+    if (!systemObserver) {
+      systemObserver = setInterval(() => systemListener(updateTheme), 500);
```

Comment:
So the only way to detect changes is to poll for changes in that css var?

Reply:
Not the only way. My experience implementing the `window.matchMedia()` was of to no avail as mentioned above.

>Acquiring the current system theme color mode using window.matchMedia() was quite problematic as the event listener would not pick up anything. Thus, I decided to use CSS media feature in combination with CSS variables, event listeners, and the new browserSettings.overrideContentColorScheme component from the browserSettings WebExtension API.

Maybe you might have better luck implementing this than me.

Reply:
Got it, I don't think it should be that big of a deal, if we have something indefinitely polling. If it starts to impact the performance of the browser, then maybe we can revisit this.

Reply:
> Got it, I don't think it should be that big of a deal, if we have something indefinitely polling. If it starts to impact the performance of the browser, then maybe we can revisit this.

@Unthrottled Well keep in mind the polling only starts when `druthers` or `system` *(A.K.A. device)* is active. otherwise the polling will stop.